src/widget/wcolorpicker.cpp

```diff
@@ -0,0 +1,95 @@
+#include "widget/wcolorpicker.h"
+
+#include <QGridLayout>
+#include <QMapIterator>
+#include <QPushButton>
+#include <QStyle>
+
+#include "util/color/color.h"
+#include "util/parented_ptr.h"
+
+namespace {
+const int kNumColumns = 4;
+}
+
+WColorPicker::WColorPicker(QWidget* parent)
+        : QWidget(parent) {
+    // If another title would be more appropriate in some context, setTitle
+    // can be called again after construction.
+    QGridLayout* pLayout = new QGridLayout();
+    pLayout->setMargin(0);
+    pLayout->setContentsMargins(0, 0, 0, 0);
+
+    m_pStyle = QStyleFactory::create(QString("fusion"));
+    int row = 0;
+    int column = 0;
+    for (const auto& pColor : Color::kPredefinedColorsSet.allColors) {
+        if (*pColor == *Color::kPredefinedColorsSet.noColor) {
```

Comment:
This is from: https://github.com/mixxxdj/mixxx/blob/a7a79385cac013ab78150c6043629be0ef30fa5f/src/widget/colormenu.cpp

It'd rather not introduce even more merge conflicts that @ferranpujolcamins, it think it will be dealt with in #2345 anyway.

Reply:
Ok